astropy/table/table.py

```diff
@@ -273,6 +273,15 @@ def __init__(self, data=None, masked=None, names=None, dtype=None,
 
         default_names = None
 
+        if hasattr(data, '__astropy_table__'):
+            # Data object implements the __astropy_table__ interface method.
+            # Calling that method returns an appropriate instance of
+            # self.__class__ and respects the `copy` arg.  The returned
+            # Table object should NOT then be copied (though the meta
+            # will be deep-copied anyway).
+            data = data.__astropy_table__(self.__class__, copy)
```

Comment:
OK, but it seems like the implication is that positional args are being disallowed in astropy internal methods.